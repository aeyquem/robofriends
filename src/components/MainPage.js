import React, { Component, Fragment } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import '../containers/App.css';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundary'
import CounterButon from '../components/CounterButton'


class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }

    filterRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        });
    }

    render() {
        const { searchField, onSearchChange, robots, isPending } = this.props;

        if (isPending) {
            return <h1>Loading...</h1>
        }
        else {
            return (
                <Fragment>
                    <h1 className="tc f2">RoboFriends</h1>
                    <CounterButon />
                    <SearchBox searchChange={onSearchChange}></SearchBox>
                    <Scroll>
                        <ErrorBoundary>
                            <CardList robots={this.filterRobots()} />
                        </ErrorBoundary>
                    </Scroll>
                </Fragment>
            );
        }
    }
}

export default MainPage;