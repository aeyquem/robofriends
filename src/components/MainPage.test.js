import React from 'react'
import { shallow } from 'enzyme'
import MainPage from './MainPage'

let wrapper;
beforeEach(() => {
    const mockProps = {
        searchField: '',
        robots: [],
        isPending: false,
        error: '',
        onRequestRobots: jest.fn()
    };

    wrapper = shallow(<MainPage {...mockProps} />)
})

it('Renders', () => {
    expect(wrapper).toMatchSnapshot();
})

it('Filter robots', () => {

    const mockProps2 = {
        searchField: 'l',
        robots: [
            {
                id: 1,
                name: 'Leanne Graham',
                username: 'Bret',
                email: 'Sincere@april.biz'
            },
        ],
        isPending: false,
        error: '',
        onRequestRobots: jest.fn()
    };

    const wrapper2 = shallow(<MainPage {...mockProps2} />)

    expect(wrapper.instance().filterRobots()).toEqual([]);
    expect(wrapper2.instance().filterRobots()).toEqual([
        {
            id: 1,
            name: 'Leanne Graham',
            username: 'Bret',
            email: 'Sincere@april.biz'
        },
    ]);
})


it('Shows loading', () => {

    const mockProps3 = {
        searchField: '',
        robots: [],
        isPending: true,
        error: '',
        onRequestRobots: jest.fn()
    };

    const wrapper3 = shallow(<MainPage {...mockProps3} />)

    expect(wrapper3).toMatchSnapshot();
})