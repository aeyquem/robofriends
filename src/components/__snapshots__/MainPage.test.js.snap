// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter robots 1`] = `ShallowWrapper {}`;

exports[`Renders 1`] = `ShallowWrapper {}`;

exports[`Shows loading 1`] = `ShallowWrapper {}`;
